### PR TITLE
Xml doc comment cleanup

### PR DESCRIPTION
**Pull request by abelbraaksma**
Just some cleanup from discussions in #219.

@bartelink feel free to commit to this branch if you have better wording for some doc comments. You mentioned yesterday you wanted to give some of it another look.

### CODE REVIEW

**Comment by abelbraaksma** [comment]
I just noticed that some functions don't have a `<returns>` element in the doc xml (like `length` or `lengthBy` functions). I'll check that tomorrow (well, later in the AM of today, that is).

**Comment by bartelink** [comment]
Hm not sure I have push rights (have pushed to https://github.com/bartelink/FSharp.Control.TaskSeq/tree/doc-cleanup instead for now)

**Comment by abelbraaksma** [comment]
@bartelink can you try again? I upped your rights, you should be allowed to push (but not sure that allows you to push on other branches). Otherwise, we can always take a merge from your fork.

File: src/FSharp.Control.TaskSeq/TaskSeqInternal.fs [hunk 1]
```diff
@@ -908,13 +891,11 @@ module internal TaskSeqInternal =
                 go <- step
         }
 
-        interface IAsyncDisposable with
-            override _.DisposeAsync() =
-                if not (isNull _rwLock) then
+        interface IDisposable with
```
Comment (bartelink) [inline src/FSharp.Control.TaskSeq/TaskSeqInternal.fs:894]: this shuts up the code suggestion about removing the `new`, and makes me more confident that it'll actually Dispose
@abelbraaksma So, what kind of Chesterton's fence have I fallen upon?

Comment (abelbraaksma) [reply src/FSharp.Control.TaskSeq/TaskSeqInternal.fs:856]: This is a bug in F# (i.e., not raising the warning), we should not fix the mistake in the library by implementing an extra interface, we should fix F#@. Some very old versions of this code used to have it.

Comment (abelbraaksma) [reply src/FSharp.Control.TaskSeq/TaskSeqInternal.fs:856]: But I'll double-check the "best practices" here. I think it's bad practice to offer synchronous disposal of async resources, but let's check.

This fix can lead to backward compat issues, so we need to take it out for sure.

File: src/FSharp.Control.TaskSeq/TaskSeqInternal.fs [hunk 2]
```diff
@@ -732,134 +732,79 @@ module internal TaskSeqInternal =
...
-                while more do
-                    let value = e.Current
-                    let! passed = predicate value
-                    more <- passed
-
-                    // yield regardless of predicate
-                    yield value
+                        cont <- hasMore
+                    | false ->
+                        match whileKind with
```
Comment (abelbraaksma) [inline src/FSharp.Control.TaskSeq/TaskSeqInternal.fs:759]: I understand the sentiment, but I prefer not having this business logic in the body as it incurs an extra penalty and allocation (remember that value types are boxed as params in an object in this case). 

So, in general, I try to keep the DUs that declare the function type as far as reasonable on the fringes of the body, and I have been thinking of getting rid of them completely, or at least have them outside of the body of the `taskSeq`.

**Review by abelbraaksma — COMMENTED** [review]
Just this one for now, will look more into the rest.

File: src/FSharp.Control.TaskSeq/TaskSeqInternal.fs (hunk 2 above)
Comment (bartelink) [reply src/FSharp.Control.TaskSeq/TaskSeqInternal.fs:759]: I assigned the to a bool before, but the match needed something like 5 lines so I went for clear code - will move it back

Comment (bartelink) [reply src/FSharp.Control.TaskSeq/TaskSeqInternal.fs]: fixed

File: src/FSharp.Control.TaskSeq/TaskSeqInternal.fs (hunk 1 above)
Comment (abelbraaksma) [reply src/FSharp.Control.TaskSeq/TaskSeqInternal.fs]: See #227, that's where this went.

**Comment by abelbraaksma** [comment]
These changes form an integral part of this continuation-PR: #234. With few exceptions, all changes have been accepted.

**Pull request closed without merging**